### PR TITLE
bpo-33525: Add env type checking when spawn called

### PR DESCRIPTION
**Pull request by Licht-T**
This fixes [bpo-33525](https://bugs.python.org/issue33525).

https://bugs.python.org/issue33525

### CODE REVIEW

File: Lib/os.py [hunk 1]
```diff
@@ -847,6 +848,14 @@ def _spawnvef(mode, file, args, env, func):
             raise TypeError('argv must be a tuple or a list')
         if not args or not args[0]:
             raise ValueError('argv first element cannot be empty')
+
+        sig = inspect.signature(func)
```
Comment (vstinner) [inline Lib/os.py:852]: I don't think that it's appropriate to use inspect here. I would prefer to see the check in the caller function instead. Add an helper function if you want to factorize the code.

**Review by vstinner — CHANGES_REQUESTED** [review]

File: Lib/os.py (hunk 1 above)
Comment (Licht-T) [reply Lib/os.py:852]: @vstinner So you mean the validation should be placed on function which call this `_spawnvef` function, right?

Comment (vstinner) [reply Lib/os.py:852]: Move the test into spawnve() and spawnvpe().

**Comment by csabella** [comment]
@Licht-T, please apply the changes that were requested in the code review.  If you're not interested in following up with this, feel free to close the pull request.  Thank you!

**Comment by AlexWaygood** [comment]
@Licht-T, thanks for the PR! However, I'm closing this PR, for the following reasons:

- Changes were requested by a core dev four years ago, but have not been made.
- Multiple checks are failing in CI
- There's now a merge conflict.

If you're still interested in working on this issue, please feel free to open a new PR :) alternatively ping me and I'll happily reopen the PR

**Pull request closed without merging**